fastestimator/backend/binary_crossentropy.py

```diff
@@ -72,6 +90,11 @@ def binary_crossentropy(y_pred: Tensor, y_true: Tensor, from_logits: bool = Fals
             ce = torch.nn.BCEWithLogitsLoss(reduction="none")(input=y_pred, target=y_true.view(y_pred.size()))
         else:
             ce = torch.nn.BCELoss(reduction="none")(input=y_pred, target=y_true.view(y_pred.size()))
+
+        if class_weights is not None:
+            sample_weights = y_true.detach().clone().cpu().apply_(lambda x: class_weights.get(int(x), 1.0))
```

Comment:
why in-place apply instead of just multiplying? if you detach y_true won't this mess with gradients later for training?

Reply:
y_true.detach().clone() will return a new tensor with the same data. There won't be any change in y_true. Applying in-place will operate on all elements of y_true which would allow different weights even for mask-like y_true.